### PR TITLE
Adding ROCm base runtime image for training

### PR DESCRIPTION
**Pull request by Fiona-Waters**
## Description

[RHOAIENG-13241](https://issues.redhat.com/browse/RHOAIENG-13241)
Added a base image for training that includes:
* UBI 9
* Python 3.11
* ROCm 6.1
* PyTorch 2.4.1

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: images/runtime/training/rocm/Pipfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+verify_ssl = true
+name = "pypi"
+
+[packages]
+peft = "*"
+datasets = ">=2.15.0"
+transformers = ">=4.41.2"
+numpy = "<2.0.0,>=1.23.5"
+accelerate = ">=0.34.2"
+torch = "==2.4.1"
```
Comment (astefanutti) [inline images/runtime/training/rocm/Pipfile:12]: The ROCm enabled PyTorch should usual come from a specific index URL, is it the case here?

Comment (Fiona-Waters) [reply images/runtime/training/rocm/Pipfile:12]: I wasn't aware of that. Let me look into it. Thanks

Comment (Fiona-Waters) [reply images/runtime/training/rocm/Pipfile:12]: torch 2.4.1 is compatible with running `pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/rocm6.` It also installs pillow-10.2.0 torchaudio-2.4.1+rocm6.1 and torchvision-0.19.1+rocm6.1 so maybe we should also include those. wdyt?

Comment (Fiona-Waters) [reply images/runtime/training/rocm/Pipfile:12]: Do we want to install it like this, or include it in Pipfile?

Comment (Fiona-Waters) [reply images/runtime/training/rocm/Pipfile:12]: I think we need to include it like the example you showed. Will update.

Comment (astefanutti) [reply images/runtime/training/rocm/Pipfile:12]: It's likely there is a way to do that in the Pipfile, I think that'd be the preferred way.

Comment (Fiona-Waters) [reply images/runtime/training/rocm/Pipfile]: Updated to include torch from index in pipfile.

Comment (astefanutti) [reply images/runtime/training/rocm/Pipfile]: Awesome!

**Review by astefanutti — APPROVED** [review]
Thanks!

**Pull request merged**